PluginGenerator/PluginDefinition.cs

```diff
@@ -28,19 +27,7 @@ public PluginManifest()
         /// <summary>
         /// Provide a unique identifier for the plugin
         /// </summary>
-        /// <remarks>By convention these are lower-case so the returned value will always be lower-cased.</remarks>
-        public string Key
-        {
-            get { return this.pluginKey; }
-            set
-            {
-                this.pluginKey = value;
-                if (this.pluginKey != null)
-                {
-                    this.pluginKey = this.pluginKey.ToLowerInvariant();
-                }
-            }
-        }
+        public string Key { get; set; }
```

Comment:
The validation is being performed elsewhere; it doesn't need to be performed here too.